.github/workflows/merge.yaml

```
@@ -0,0 +1,235 @@
+name: "GitHub CI merge checks"
+
+on:
+  push:
+    branches: [ "auto", "github-actions-merge-tests" ]
+
+env:
+  PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
+  CHECKOUT_FETCH_DEPTH: 1001
+
+jobs:
+  linux-matrix-tests:
+
+    strategy:
+      matrix:
+        os:
+          - centos-stream-9
+          - debian-stable
+          - debian-testing
+          - debian-unstable
+          - fedora-39
+          - fedora-40
+          - fedora-rawhide
+          - gentoo
+          - opensuse-leap
+          - opensuse-tumbleweed
+          - ubuntu-focal
+          - ubuntu-jammy
+          - ubuntu-noble # EOL 2036-04
+          - ubuntu-oracular # EOL 2025-07
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
+        layer:
+          - { name: layer-00-default, nick: default }
+          - { name: layer-01-minimal, nick: minimal }
+          - { name: layer-02-maximus, nick: maximus }
+        # exclude:
+        #  if any combinations needs to be excluded, do it here
+        #   - { os: macos-14, compiler: { CC: gcc, CXX: g++ } }
+
+    runs-on: ubuntu-latest
+    container:
+      image: squidcache/buildfarm-${{ matrix.os }}:latest
```

"latest" feels like the wrong tag. We should use images that were cleared/tested for Squid Project use. The latest image may not be one of those! In other words, how will you test image changes/upgrades without affecting existing/pending PRs if GitHub Actions are supposed to use the _latest_ image?

I am not a docker expert, but please consider something like this instead:

```suggestion
      image: squidcache/buildfarm-${{ matrix.os }}:production
```

or

```suggestion
      image: squidcache/buildfarm-${{ matrix.os }}:stable
```

or even something like this if we foresee using different image variations for master and v6:

```suggestion
      image: squidcache/buildfarm-${{ matrix.os }}:for-master
```

N.B. I do not know what Jenkins is using, but I do not think that should matter for this PR. We do not have to replicate all Jenkins setup problems here.

---

Our current image promotion process calls for :stable. However, opensuse-tumbleweed stable is currently broken, latest is working. I am currently testing latest for promotion. Adding a note to switch to stable before merge

---

AFAICT, we are talking about two unrelated aspects:

* I am talking about the freshest docker image produced by you (i.e. the very last image that was produced). This configuration should not use the freshest image. We should use the image that was _validated_ to work best. In some cases, it will not be the freshest. I assume that "latest" spelling implies "freshest", but perhaps that is not true.
 
* You might be talking about some designation related to OS versions. For example, right now, the latest Ubuntu release is X.Y. This change request concern is _not_ about OS versions and such!
 
Who controls those image suffixes? I assumed it was us.

How will you upgrade from one docker image to another (while staying on the same OS version and just improving/changing something inside the published/shared image)?

---

> AFAICT, we are talking about two unrelated aspects:
> 
> * I am talking about the freshest docker image produced by you (i.e. the very last image that was produced). This configuration should not use the freshest image. We should use the image that was _validated_ to work best. In some cases, it will not be the freshest. I assume that "latest" spelling implies "freshest", but perhaps that is not true.

That is, indeed, correct. "latest" is the latest image that I built and pushed to the docker hub.
I re-tag it as "stable" after I successfully test no regressions against the current test set

> * You might be talking about some designation related to OS versions. For example, right now, the latest Ubuntu release is X.Y. This change request concern is _not_ about OS versions and such!

Yes, I had understood that

> Who controls those image suffixes? I assumed it was us.

It is us.

> How will you upgrade from one docker image to another (while staying on the same OS version and just improving/changing something inside the published/shared image)?

that suffix is just a label. "latest" is by convention the most recently pushed version.
Our own promotion process is driven by [this makefile rule](https://github.com/kinkie/dockerfiles/blob/11eb9feb5cc0fc92fc091d84932a8a0af06f0abe/Makefile#L106):

---

> "latest" is the latest image that I built and pushed to the docker hub.
> I re-tag it as "stable" after I successfully test no regressions against the current test set

OK, then master (i.e. this PR) should use "stable" tag. You can use "latest" tag in your Squid repository to check new images on GitHub (if possible, I would avoid using Jenkins for those checks).